crates/router/src/core/payments/helpers.rs

```diff
@@ -1780,9 +1780,17 @@ pub async fn retrieve_card_with_permanent_token(
             .unwrap_or_default()
             .card_cvc
             .unwrap_or_default(),
-        card_issuer: card.card_brand,
+        card_issuer: None,
         nick_name: card.nick_name.map(masking::Secret::new),
-        card_network: None,
+        card_network: card
+            .card_brand
+            .map(|card_brand| enums::CardNetwork::from_str(&card_brand))
+            .transpose()
+            .map_err(|e| {
+                logger::error!("Failed to parse card network {}", e);
+            })
+            .ok()
+            .flatten(),
```

Comment:
Should we do this? I feel ignoring errors isn't a very good approach for solving this

Reply:
This error itself won't occur in the first place as we would have already evaluated while storing it